### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+                 load_factor: float = 0.67
+                 ) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.occupied = 0
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(
+            self,
+            key: str | int | float | tuple,
```
Comment (vasylhnatiuk) [inline app/main.py:17]:
```suggestion
            key: Hashable,
```

File: app/main.py [hunk 2]
```diff
@@ -1,2 +1,50 @@
+from typing import Any
```
Comment (vasylhnatiuk) [inline app/main.py:1]:
```suggestion
from typing import Any, Hashable
```

File: app/main.py [hunk 3]
```diff
@@ -1,2 +1,50 @@
...
+            table_copy = self.hash_table
+            self.capacity = self.capacity * 2
+            self.hash_table = [None] * self.capacity
+            self.occupied = 0
+            for node in table_copy:
+                if node:
+                    self[node[0]] = node[1]
+
+    def __getitem__(self,
+                    key: str | int | float | tuple
```
Comment (vasylhnatiuk) [inline app/main.py:40]:
```suggestion
                    key: Hashable
```

File: app/main.py [hunk 4]
```diff
@@ -1,2 +1,50 @@
...
+        while (self.hash_table[index_]
+               and self.hash_table[index_][0] != key):
+            index_ = (index_ + 1) % self.capacity
+        if (self.hash_table[index_]
+                and self.hash_table[index_][0] == key):
+            self.hash_table[index_][1] = value
+        if not self.hash_table[index_]:
+            self.occupied += 1
+            self.hash_table[index_] = [key, value]
+        if self.occupied >= self.capacity * self.load_factor:
```
Comment (vasylhnatiuk) [inline app/main.py:30]: Create a separate method, `__resize` with bellow code . And call it here

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Good job!

**Pull request closed without merging**